### PR TITLE
Internal server proposal

### PR DESCRIPTION
**Pull request by delvedor**
As titled.

I've exposed the `listen` method and the `server` object.
Should we test the `https` server creation?

### CODE REVIEW

**Comment by mcollina** [comment]
We should also expose `fastify.address()`.

Be sure to tie in `server.listen` into a `boot.ready()` thing, so all plugins are loaded.

Also.. docs :D

**Comment by delvedor** [comment]
I'm already exposing `fastify.server`, which contains `.address()`, why expose `.address()` alone? Isn't a little bit redundant?

File: README.md [hunk 1]
```diff
@@ -96,18 +96,38 @@ Core.
...
 
-  console.log(`server listening on ${server.address().port}`)
+  console.log(`server listening on ${fastify.server.address().port}`)
 })
 ```
 
+<a name="server"></a>
+### fastify.server
+
+The Node core [server object](https://nodejs.org/api/net.html#net_class_net_server).
```
Comment (mcollina) [inline README.md]: this is either http or https

File: README.md [hunk 2]
```diff
@@ -96,18 +96,38 @@ Core.
...
+
+If you need an `HTTPS` server, pass an option object with the keys to the *fastify* constructor.
+```js
+const options = {
+  https: {
+    key: fs.readFileSync('test/fixtures/keys/agent2-key.pem'),
+    cert: fs.readFileSync('test/fixtures/keys/agent2-cert.pem')
+  }
+}
+const fastify = require('fastify')(options)
```
Comment (mcollina) [inline README.md]: this should be moved in the construstor

Comment (mcollina) [inline README.md]: we should also link to the HTTPS module in core for all the options

Comment (mcollina) [inline README.md]: We should also mention about the ready() behavior

File: README.md [hunk 3]
```diff
@@ -96,18 +97,44 @@ Core.
...
+### fastify.ready(callback)
+
+Function called when all the plugins has been loaded.  
+Emitted by [boot-in-the-arse](https://github.com/mcollina/boot-in-the-arse).
+
+<a name="listen"></a>
+### fastify.listen(port, [callback])
+
+Starts the server on the given port after all the plugins are loaded.  
+The callback is the same as the Node core.
```
Comment (mcollina) [inline README.md]: we should state we are waiting for `ready` to happen.

**Pull request merged**